correct_periscope_drift.py

```diff
@@ -0,0 +1,431 @@
+    evt_dec = evt_dec[ok_times]
+    evt_times = evt_times[ok_times]
+
+    if len(evt_times) < opt['src_min_counts']:
+        v1("Warning only {} counts in src region.  {} minimum suggested 'src_min_counts'".format(
+                len(evt_times), opt['src_min_counts']))
+
+    ax_data = {}
+    ax_map = {'yag': 'dy',
+              'zag': 'dz'}
+
+    ax_data['yag'], ax_data['zag'] = get_event_yag_zag(evt_ra, evt_dec,
+                                                       evt_ra_nom, evt_dec_nom, evt_roll_nom)
+
+    # Store comments to print in block after all of the sherpa fit output
+    fit_comments = []
+    plot_list = []
+
+    for data_id, ax in enumerate(['yag', 'zag']):
+        fit_data = ax_data[ax] - np.mean(ax_data[ax])
+        mp = fit(fit_data, evt_times, data_id, opt)
+
+        bin_centers, bin_mean, bin_std = time_bins(evt_times, fit_data)
+
+        add_window()
+        add_curve((bin_centers - evt_times[0]) / 1000., bin_mean, [bin_std, +bin_std],
+                  ["line.style", "none", "symbol.style", "none", "err.style", "cap"])
+        add_curve(mp.x / 1000., mp.y, ["symbol.style", "none"])
+        set_plot_xlabel("Observation elapsed/delta time (ks)")
+        set_plot_ylabel("Position offset from mean, {} (arcsec)".format(ax))
+        set_plot_title("Fit of {} data (with time-binned event offsets)".format(ax))
+        fit_plot = "{}_fit_{}.png".format(opt['corr_plot_prefix'], ax)
+        if os.path.exists(fit_plot) and opt['clobber']:
+            os.unlink(fit_plot)
+        plot_list.append(fit_plot)
+        print_window(fit_plot)
+
+        add_window()
+        data_plot = "{}_data_{}.png".format(opt['corr_plot_prefix'], ax)
+        ui.plot_fit(data_id)
+        if os.path.exists(data_plot) and opt['clobber']:
+            os.unlink(data_plot)
+        set_plot_xlabel("Observation elapsed/delta time (s)")
+        set_plot_ylabel("Position offset from mean, {} (arcsec)".format(ax))
+        set_plot_title("Raw data and fit in {}".format(ax))
+        plot_list.append(data_plot)
+        print_window(data_plot)
+
+        asol_corr = np.interp(asol_times, mp.x + evt_times[0], mp.y)
+        asol_col_to_fix = asol.get_column(ax_map[ax])
+        fit_comments.append("Events show drift range of {:.2f} arcsec in {} axis".format(
+                np.max(asol_corr) - np.min(asol_corr), ax))
+        fit_comments.append("Max absolute correction of {:.2f} arcsec for {} axis".format(
+                np.max(np.abs(asol_corr)), ax))
+
+        asol_col_to_fix.values += (asol_corr / 20)
+
+    v1("-" * 60)
+    v1("Fit results")
+    for c in fit_comments:
+        v1("\t{}".format(c))
+    v1("-" * 60)
+    v2("Writing out corrected aspect solution file to {}".format(opt['outfile']))
+    v2("\tTo review fit see correction plots in:")
+    for p in plot_list:
+        v2("\t\t{}".format(p))
+
+    # Actually write out the new aspect solution file
+    asol.write(opt['outfile'], clobber=opt['clobber'])
```

Comment:
Humm ... the `DY_AVG` and `DZ_AVG` in the header of the event file is set to mean value of the values in the asol file(s).  It may be that we need to add a step to the thread that recomputes those after foo_process_events has been run.